tools/ThermoRawFileParser/thermo_converter.xml

```diff
@@ -1,60 +1,193 @@
+
     &&
-    #if $output_format == '0':
-        mv ./input.mgf '$output'
-    #else:
-        mv ./input.mzML '$output'
+
+    #if len($input) == 1:
+        #if $output_format == "0":
+            mv ./output_folder/input.mgf ./output_file.out &&
+        #else if $output_format == "1":
+            mv ./output_folder/input.mzML ./output_file.out &&
+        #else if $output_format == "2":
+            mv ./output_folder/input.mzML ./output_file.out &&
+        #end if
+
+        #if $output_metadata_selector != "off":
+            #if $output_metadata_selector == "0":
+                mv ./output_folder/input-metadata.json ./input-metadata.txt &&
+            #else if $output_metadata_selector == "1":
+                mv ./output_folder/input-metadata.txt ./input-metadata.txt &&
+            #end if
+        #end if
     #end if
+
+    cat $temp_stderr 2>&1;
 ]]>
     </command>
     <inputs>
-        <param name="input" type="data" format="thermo.raw" label="Thermo RAW file" help="" />
-        <param name="metadata" type="boolean" truevalue="" falsevalue="" checked="False"
-            label="Output metadata" help="" />
-        <param name="output_format" type="select" label="Choose an output format">
+        <param name="input" type="data" format="thermo.raw" label="Thermo RAW file" help="" multiple="true"
+          optional="False" />
+
+        <param name="output_format" type="select" label="Output format">
             <option value="0">mgf</option>
             <option value="1" selected="True">mzml</option>
+            <option value="2">Indexed mzml</option>
         </param>
+
+        <param name="zlib_boolean" type="boolean" truevalue="" falsevalue="-z" checked="true"
+            label="Use zlib compression for the m/z ratios and intensities" help="" />
+
+        <param name="peakpicking_boolean" type="boolean" truevalue="" falsevalue="-p" checked="true"
+            label="Use the peak picking provided by the native thermo library" help="" />
+
+        <param name="ignore_instrument_errors_boolean" type="boolean" truevalue="-e" falsevalue="" checked="true"
+            label="Ignore missing instrument properties" help="If false, it stops the conversion if instrument properties are missing" />
+
+        <param name="output_metadata_selector" type="select" label="Output metadata" >
+            <option value="off" selected="True">No</option>
+            <option value="0">json</option>
+            <option value="1">txt</option>
+        </param>
+
     </inputs>
     <outputs>
-        <data format="mzml" name="output" from_work_dir="input.mzML" label="${tool.name} on ${on_string}">
+
+        <!-- We use simple data outputs if we just have one file, for backwards compatibility -->
+
+        <data name="output" format="mzml" from_work_dir="output_file.out" label="${tool.name} on ${on_string}">
+            <filter>(str(input)).count(',') == 0</filter> <!-- funny way of counting the number of input files! -->
```

Comment:
wired ... we should create a simple Galaxy tool that confirms this and add it to Galaxy core as a test tool ...

Reply:
Well, I am just counting the number of commas present in the input hehe. The really weird thing (and dangerous, because is misleading) is what happens with the len(input) function in the filter, which seems to work nicely when running the tool itself but it does not work with the tests (seems to crash internally but the condition is approved)... :-S